apps/agentic-chat/src/hooks/useTools.ts

```diff
@@ -1,107 +1,224 @@
 import { useAccount, useWalletClient } from 'wagmi';
-import { Address, getAddress, Hex } from 'viem';
-import { ToolCall } from '@ai-sdk/provider-utils';
+import { getAddress, Hex } from 'viem';
 import { BebopQuote } from '@agentic-chat/types';
-
+import { useAssistantTool } from '@assistant-ui/react';
+import { z } from 'zod';
 import { useState } from 'react';
-import { getAccount } from '../tools/getAccount';
-import { switchEvmChain } from '../tools/switchEvmChain';
 import { searchTokens } from '../tools/searchTokens';
-import { getBebopRate } from '../tools/bebopRate';
 import { getAllowance } from '../tools/getAllowance';
 import { approve } from '../tools/approve';
 import { sendTransaction } from '../tools/sendTransaction';
 import { toBaseUnit } from '@agentic-chat/utils';
+import { getAccount } from '../tools/getAccount';
+import { getBebopRate } from '../tools/bebopRate';
 
 const useTools = () => {
   const account = useAccount();
   const { data: walletClient } = useWalletClient();
   const [bebopQuote, setBebopQuote] = useState<BebopQuote | null>(null);
 
-  const handleToolCall = async ({
-    toolCall,
-  }: {
-    toolCall: ToolCall<string, unknown>;
-  }) => {
-    switch (toolCall.toolName) {
-      case 'getAddress': {
-        return account.address;
-      }
-
-      case 'getAccount': {
-        return getAccount(account, toolCall);
-      }
-
-      case 'switchEvmChain': {
-        return switchEvmChain(walletClient, toolCall);
-      }
-
-      case 'searchTokens': {
-        return searchTokens(toolCall);
-      }
-
-      case 'bebopRate': {
-        return getBebopRate({ toolCall, setBebopQuote });
-      }
-
-      case 'getAllowance': {
-        return getAllowance(account, toolCall);
+  useAssistantTool({
```

Comment:
Tools are now declared client-side, which is a hard requirement here since it's not possible anymore to `onToolCall()` with `useChatRuntime`